bn: Remove the BN_MUL_COMBA cpp define

arrowd opened the pull request:
It does not represent a feature that some arch may or may not possess, but instead is entirely dependent on the OPENSSL_SMALL_FOOTPRINT option.

I found having `BN_MUL_COMBA` a little more confusing when porting OpenSSL to a new architecture.

arrowd committed:
bn: Remove the BN_MUL_COMBA cpp define

It does not represent a feature that some arch may or may not possess, but
instead is entirely dependent on the OPENSSL_SMALL_FOOTPRINT option.

fwh-dc reviewed (changes requested):
This reduces complexity which is nice, but I am not sure if the macros can be set without setting OPENSSL_SMALL_FOOTPRINT.

Otherwise LGTM except the other macros should be removed as well.

arrowd commented on crypto/bn/bn_local.h:
```diff
@@ -53,7 +53,6 @@
 # ifndef OPENSSL_SMALL_FOOTPRINT
-#  define BN_MUL_COMBA
 #  define BN_SQR_COMBA
 #  define BN_RECURSION
```
Yes, I was going to go through all of them once this PR gets merged.

But if you're saying that this is generally OK, I can do all of them in one go.

arrowd committed:
bn: Remove the BN_SQR_COMBA cpp define

Just like in previous commit, this define does not represent a toggleable
feature, but is entirely dependent on the OPENSSL_SMALL_FOOTPRINT define.

arrowd committed:
bn: Remove the BN_RECURSION cpp define

Just like in previous commit, this define does not represent a toggleable
feature, but is entirely dependent on the OPENSSL_SMALL_FOOTPRINT define.

fwh-dc reviewed (dismissed):
LGTM

openssl-machine commented:
This pull request is ready to merge

t8m reviewed (approved):
I assume @fwh-dc approval still holds.
